parser.cc

```diff
@@ -1169,7 +1174,7 @@ bool Context::parse_rule(const Rule &r, bool (Context::*parse_func)(const Rule &
 	}
 
 	// Return value (success or failure of parse)
-	bool ok;
+	bool ok = false;
```

Comment:
This looks wrong.  On which code path was ok used uninitialised?  If there is one, then this will hide a bug, not fix it.